dvc/repo/__init__.py

```diff
@@ -410,6 +413,9 @@ def func(out):
             if recursive and out.path_info.isin(path_info):
                 return True
 
+            if glob:
+                return fnmatch.fnmatch(out.path_info.name, path)
```

Comment:
`name` is a basename, we should be checking for the `str(path_info)` which should also make it work with `../data/**` kind of globs.

Reply:
I changed that and added new test to check how globbing with directories in path works.